src/renamer.fs

```diff
@@ -24,7 +24,7 @@ module private RenamerImpl =
         // Whether multiple functions can have the same name (but different arity).
```

Comment:
We should use Ident for AST nodes. `availableNames` should be a `string list`.
Also, maybe:
```
        varRenames: Map<string, Ident>
        funRenames: Map<string, Map<int, Ident>>
```

Reply:
I'm planning Equals+GetHashCode for Ident

Reply:
Maps don't need hashable (it's a tree).
I don't think we want to compare Idents.